daml-script/dump/src/main/scala/com/daml/script/dump/TreeUtils.scala

```diff
@@ -94,6 +94,20 @@ object TreeUtils {
     cids
   }
 
+  def treeReferencedCids(tree: TransactionTree): Set[String] = {
+    var cids: Set[String] = Set.empty
+    traverseTree(tree) { case (_, kind) =>
+      kind match {
+        case Kind.Empty =>
+        case Kind.Exercised(value) =>
+          cids += value.contractId
+          cids ++= value.choiceArgument.foldMap(arg => valueCids(arg.sum))
+        case Kind.Created(_) =>
```

Comment:
We also need to recurse here on the create argument. A template can contain a contract id as a field.

Reply:
Good catch, added.